### DISCUSSION TITLE
Log errors for yamlls configuration with customTags

### DISCUSSION THREAD

**Discussion by LukaK** [williamboman/nvim-lsp-installer#469] [Q&A] [answered]
Hi,
I am seeing error in lsp logs even dough server is working normally. It started happening when customTags are added.

Any idea what is the issue?

Configuration:

```
M.options = {
  filetypes = {"yaml", "yml"},
  settings = {
    yaml = {
      schemas = schemastore.json.schemas(),
      -- aws cloudformation tags
      customTags = {
        "!And scalar",
        "!And map",
        "!And sequence",
        "!If scalar",
        "!If map",
        "!If sequence",
        "!Not scalar",
        "!Not map",
        "!Not sequence",
        "!Equals scalar",
        "!Equals map",
        "!Equals sequence",
        "!Or scalar",
        "!Or map",
        "!Or sequence",
        "!FindInMap scalar",
        "!FindInMap mappping",
        "!FindInMap sequence",
        "!Base64 scalar",
        "!Base64 map",
        "!Base64 sequence",
        "!Cidr scalar",
        "!Cidr map",
        "!Cidr sequence",
        "!Ref scalar",
        "!Ref map",
        "!Ref sequence",
        "!Sub scalar",
        "!Sub map",
        "!Sub sequence",
        "!GetAtt scalar",
        "!GetAtt map",
        "!GetAtt sequence",
        "!GetAZs scalar",
        "!GetAZs map",
        "!GetAZs sequence",
        "!ImportValue scalar",
        "!ImportValue map",
        "!ImportValue sequence",
        "!Select scalar",
        "!Select map",
        "!Select sequence",
        "!Split scalar",
        "!Split map",
        "!Split sequence",
        "!Join scalar",
        "!Join map",
        "!Join sequence",
      },
    }
  }
}
```

Error message:

> [ERROR][2022-02-10 02:17:45] ...lsp/handlers.lua:404	"TypeError: Cannot read properties of undefined (reading 'filter')\n    at Object.filterInvalidCustomTags (/home/luka/.local/share/nvim/lsp_servers/yaml/node_modules/yaml-language-server/out/server/src/languageservice/utils/arrUtils.js:58:23)\n    at Object.getCustomTags (/home/luka/.local/share/nvim/lsp_servers/yaml/node_modules/yaml-language-server/out/server/src/languageservice/parser/custom-tag-provider.js:51:37)\n    at Object.parse (/home/luka/.local/share/nvim/lsp_servers/yaml/node_modules/yaml-language-server/out/server/src/languageservice/parser/yamlParser07.js:26:43)\n    at YamlDocuments.ensureCache (/home/luka/.local/share/nvim/lsp_servers/yaml/node_modules/yaml-language-server/out/server/src/languageservice/parser/yaml-documents.js:201:40)\n    at YamlDocuments.getYamlDocument (/home/luka/.local/share/nvim/lsp_servers/yaml/node_modules/yaml-language-server/out/server/src/languageservice/parser/yaml-documents.js:179:14)\n    at YAMLValidation.<anonymous> (/home/luka/.local/share/nvim/lsp_servers/yaml/node_modules/yaml-language-server/out/server/src/languageservice/services/yamlValidation.js:60:74)\n    at Generator.next (<anonymous>)\n    at /home/luka/.local/share/nvim/lsp_servers/yaml/node_modules/yaml-language-server/out/server/src/languageservice/services/yamlValidation.js:13:71\n    at new Promise (<anonymous>)\n    at __awaiter (/home/luka/.local/share/nvim/lsp_servers/yaml/node_modules/yaml-language-server/out/server/src/languageservice/services/yamlValidation.js:9:12)"

**Comment by williamboman** [comment]
Hello! First of all I'd check the Node version you're using, how recent is it?

**Comment by LukaK** [comment, marked as answer]
Sure thing.

```
node --version  
v17.3.0
```

```
nvim --version 
NVIM v0.6.1
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by builduser

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info

```

```
uname -a 
Linux probe 5.16.8-arch1-1 #1 SMP PREEMPT Tue, 08 Feb 2022 21:21:08 +0000 x86_64 GNU/Linux
```

```
nvim-lsp-installer: require("nvim-lsp-installer.health").check()
========================================================================
## nvim-lsp-installer report
  - OK: neovim version >= 0.6.0
  - WARNING: **Composer**: not available
  - WARNING: **PHP**: not available
  - WARNING: **julia**: not available
  - OK: **tar**: `tar (GNU tar) 1.34`
  - OK: **gzip**: `gzip 1.11`
  - OK: **curl**: `curl 7.81.0 (x86_64-pc-linux-gnu) libcurl/7.81.0 OpenSSL/1.1.1m zlib/1.2.11 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.2 libpsl/0.21.1 (+libidn2/2.3.0) libssh2/1.10.0 nghttp2/1.46.0`
  - OK: **wget**: `GNU Wget 1.21.2 built on linux-gnu.`
  - OK: **node**: `v17.3.0`
  - OK: **Ruby**: `ruby 3.0.3p157 (2021-11-24 revision 3fb7d2cadc) [x86_64-linux]`
  - OK: **Go**: `go version go1.17.6 linux/amd64`
  - OK: **python3_host_prog**: `Python 3.7.9`
  - OK: **bash**: `GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)`
  - OK: **sh**: `Ok`
  - OK: **python3**: `Python 3.10.2`
  - OK: **RubyGem**: `3.3.5`
  - OK: **java**: `Ok`
  - OK: **npm**: `8.4.1`
  - OK: **javac**: `javac 17.0.1`
  - OK: **pip3**: `pip 20.3.4 from /usr/lib/python3.10/site-packages/pip (python 3.10)`
```

But healthchecks for neovim providers are failing. Could that be an issue?
```
provider: health#provider#check
========================================================================
  - ERROR: Failed to run healthcheck for "provider" plugin. Exception:
    function health#check[20]..health#provider#check[6]..<SNR>97_check_node, line 20
    Vim(if):E117: Unknown function: provider#node#can_inspect
```

**Reply by williamboman** [reply]
I can't really repro this. Have you tried installing the latest version of the server? When does the error happen? Does it happen when the server starts, or is it when you do something in specific?

**Reply by LukaK** [reply]
Seems like I cant repro this as well after yesterdays system update. But server was latest version and it happened when the server was attached to a buffer (opening a file). But thank you all the same! All the best!
